synapseclient/core/upload/multipart_upload_async.py

```diff
@@ -403,6 +398,23 @@ async def _orchestrate_upload_part_tasks(
                         part_number = task_result.part_number
                         part_size = task_result.part_size
                         part_md5_hex = task_result.md5_hex
+                    elif (
+                        isinstance(task_result, AddPartResponse)
+                        and task_result.add_part_state != "ADD_SUCCESS"
```

Comment:
Are you concerned about the response object changing from `ADD_SUCCESS` to another value?

Reply:
From the slack help thread, this is how the Web interface also handles this - We will keep this aligned.